configure.ac

```diff
@@ -1423,6 +1423,10 @@ if test "$use_boost" = "yes"; then
   if test "$suppress_external_warnings" != "no"; then
     BOOST_CPPFLAGS=SUPPRESS_WARNINGS($BOOST_CPPFLAGS)
   fi
+
+  if test "$enable_debug" = "yes"; then
+    AX_CHECK_PREPROC_FLAG([-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE], [DEBUG_CPPFLAGS="$DEBUG_CPPFLAGS -DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE"], [], [$CXXFLAG_WERROR])
```

Comment:
`AX_CHECK_PREPROC_FLAG` doesn't really do anything special for `-D` options. Maybe just add it to `DEBUG_CPPFLAGS` unconditionally?

Reply:
We are currently using `AX_CHECK_PREPROC_FLAG` for all other `-D` options we pass to our `DEBUG_CPPFLAGS`, so I think I'll leave this as-is.

Reply:
Now added unconditionally given we've got other unconditional flags.